src/Serilog/Capturing/PropertyBinder.cs

```diff
@@ -83,11 +92,26 @@ EventProperty[] ConstructPositionalProperties(MessageTemplate template, object?[
         return result;
     }
 
-    EventProperty[] ConstructNamedProperties(MessageTemplate template, object[] messageTemplateParameters)
+#if NETSTANDARD2_1 || NET5_0_OR_GREATER
+    private readonly static System.Buffers.ArrayPool<EventProperty> _eventPropertyPool = System.Buffers.ArrayPool<EventProperty>.Shared;
+#endif
+
+    internal static void Return(EventProperty[] array)
+    {
+#if NETSTANDARD2_1 || NET5_0_OR_GREATER
```

Comment:
i know @nblumhardt refers named constants to represent features. so perhaps add a 

`FEATURE_ARRAYPOOL`